From 377ec5f620561eab0e7a0e9ab4faeb8415390b89 Mon Sep 17 00:00:00 2001
From: Maital Hahn <maitalm@ti.com>
Date: Thu, 17 Mar 2016 15:15:57 +0200
Subject: [PATCH 06/18] wl18xx: multi-role AP & GO doesn't work

Added support for multi-role AP & GO, by adding this combination
to the supported combinations in the wilink driver.

Signed-off-by: Maital Hahn <maitalm@ti.com>
---
 drivers/net/wireless/ti/wl18xx/main.c | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/drivers/net/wireless/ti/wl18xx/main.c b/drivers/net/wireless/ti/wl18xx/main.c
index 496b9b63..e40f6d9 100644
--- a/drivers/net/wireless/ti/wl18xx/main.c
+++ b/drivers/net/wireless/ti/wl18xx/main.c
@@ -1916,6 +1916,28 @@ wl18xx_iface_combinations[] = {
 					BIT(NL80211_CHAN_HT20) |
 					BIT(NL80211_CHAN_HT40MINUS) |
 					BIT(NL80211_CHAN_HT40PLUS),
+	},
+	{
+		.max_interfaces = 3,
+		.limits = wl18xx_iface_ap_cl_limits,
+		.n_limits = ARRAY_SIZE(wl18xx_iface_ap_cl_limits),
+		.num_different_channels = 2,
+	},
+	{
+		.max_interfaces = 3,
+		.limits = wl18xx_iface_ap_cl_limits,
+		.n_limits = ARRAY_SIZE(wl18xx_iface_ap_cl_limits),
+		.num_different_channels = 1,
+		.radar_detect_widths =  BIT(NL80211_CHAN_NO_HT) |
+								BIT(NL80211_CHAN_HT20) |
+								BIT(NL80211_CHAN_HT40MINUS) |
+								BIT(NL80211_CHAN_HT40PLUS),
+	},
+	{
+		.max_interfaces = 3,
+		.limits = wl18xx_iface_ap_go_limits,
+		.n_limits = ARRAY_SIZE(wl18xx_iface_ap_go_limits),
+		.num_different_channels = 1,
 	}
 };
 
-- 
1.9.1

